### DISCUSSION TITLE
Rookie question : what is the best way to handle database clients (Supabase, Firebase, AppWrite...) ?

### DISCUSSION THREAD

**Discussion by Zebnastien** [nuxt/nuxt#24915] [General]
Some database tools allow JS clients to query the database through their own APIs.

My question is simple: are there any advantages to using these clients on Nuxt Server (Nitro apis) and querying them with useFetch or is it better to query them directly without an intermediary api?

What are the recommanded practices with Nuxt?

Thanks :)

**Comment by TheAlexLichter** [comment]
There are some integrations, e.g. for [Nuxt and Supabase](https://github.com/nuxt-modules/supabase). If an integration exists, I'd prefer them over rolling your own.

Calling on the client vs. on the server depends what you want to do with the data.

**Reply by Zebnastien** [reply]
Hey! Thank you for your reply.
Coincidentally, I just discovered your YouTube channel last week. Nice work, hope to see more soon!

**Reply by TheAlexLichter** [reply]
@Zebnastien Thanks a lot! More coming (one video every Friday 🙌🏻)
